mutator-common/src/display_report.rs

```diff
@@ -0,0 +1,291 @@
+//! A module for displaying reports in a nice fashion.
+// Copyright © Eiger
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::report::Report;
+use crate::report::MutantStats;
+use anyhow::{Context, Result};
+use diffy::Line;
+use prettytable::{
+    color,
+    format::{self, Alignment, LinePosition, LineSeparator},
+    Attr, Cell, Row, Table,
+};
+use std::{
+    collections::{BTreeMap, HashSet},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+const COLOR_HAPPY: Option<Attr> = Some(Attr::ForegroundColor(color::GREEN));
+const COLOR_WARN: Option<Attr> = Some(Attr::ForegroundColor(color::BRIGHT_YELLOW));
+const COLOR_CRITICAL: Option<Attr> = Some(Attr::ForegroundColor(color::RED));
+const COLOR_NONE: Option<Attr> = None;
+
+/// Filter for modules to include in the report.
+#[derive(Default, Debug, Clone, PartialEq)]
+pub enum ModuleFilter {
+    #[default]
+    All,
+    Selected(Vec<String>),
+}
+
+impl FromStr for ModuleFilter {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "all" => Ok(ModuleFilter::All),
+            _ => Ok(ModuleFilter::Selected(
+                s.split(&[';', '-', ',']).map(String::from).collect(),
+            )),
+        }
+    }
+}
+
+impl ModuleFilter {
+    fn get_all_files_containing_the_modules(&self, report: &Report) -> HashSet<PathBuf> {
```

Comment:
Using a `HashSet` will give non-deterministic ordering when you iterate over the collection?

This may mean running the same tool on the same project with everything else being same could still produce different looking results?

Reply:
True! Let's use BTreeSet then here and then we get a deterministic ordering of the files.
Done